File: api/create-issue.js
export default async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  const { url } = req.body;

  if (!url || !url.includes('instagram.com')) {
    return res.status(400).json({ error: 'Invalid Instagram URL' });
  }

  const token = process.env.GITHUB_TOKEN;
  if (!token) {
    return res.status(500).json({ error: 'GitHub token not configured' });
  }

  try {
    const reelId = url.split('/').slice(-2, -1)[0] || 'reel';

    const response = await fetch(
      'https://api.github.com/repos/xopromo/telethon/issues',
      {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Accept': 'application/vnd.github+json',
          'X-GitHub-Api-Version': '2022-11-28',
        },
        body: JSON.stringify({
          title: `Download Reel: ${reelId}`,
          body: `**Instagram URL:**\n${url}\n\n_This issue will be processed automatically by the workflow._`,
          labels: ['reel-download']
        })
      }
    );

    if (!response.ok) {
      throw new Error(`GitHub API error: ${response.statusText}`);
    }

    const issue = await response.json();

    return res.status(201).json({
      success: true,
      issue_url: issue.html_url,
      issue_number: issue.number,
      message: '✅ Issue created! GitHub Actions will process it now.'
    });

  } catch (error) {
    return res.status(500).json({
      success: false,
      error: error.message || 'Failed to create issue'
    });
  }
}

File: api/create-issue.py
from http.server import BaseHTTPRequestHandler
import json
import urllib.request
import os


class handler(BaseHTTPRequestHandler):

    def _cors(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')

    def _json(self, status, data):
        body = json.dumps(data).encode()
        self.send_response(status)
        self._cors()
        self.send_header('Content-Type', 'application/json')
        self.end_headers()
        self.wfile.write(body)

    def do_OPTIONS(self):
        self.send_response(200)
        self._cors()
        self.end_headers()

    def do_POST(self):
        length = int(self.headers.get('Content-Length', 0))
        body = self.rfile.read(length)

        try:
            data = json.loads(body)
        except Exception:
            self._json(400, {'error': 'Invalid JSON'})
            return

        url = data.get('url', '').strip()
        if not url or 'instagram.com' not in url:
            self._json(400, {'error': 'Invalid Instagram URL'})
            return

        token = os.environ.get('GITHUB_TOKEN')
        if not token:
            self._json(500, {'error': 'GITHUB_TOKEN not configured'})
            return

        reel_id = url.rstrip('/').split('/')[-1] or 'reel'

        payload = json.dumps({
            'title': f'Download Reel: {reel_id}',
            'body': f'**Instagram URL:**\n{url}\n\n_Processed by GitHub Actions workflow._',
            'labels': ['reel-download'],
        }).encode()

        req = urllib.request.Request(
            'https://api.github.com/repos/xopromo/telethon/issues',
            data=payload,
            headers={
                'Authorization': f'Bearer {token}',
                'Accept': 'application/vnd.github+json',
                'X-GitHub-Api-Version': '2022-11-28',
                'Content-Type': 'application/json',
            },
        )

        try:
            with urllib.request.urlopen(req) as resp:
                issue = json.loads(resp.read())
            self._json(201, {
                'success': True,
                'issue_url': issue['html_url'],
                'issue_number': issue['number'],
            })
        except urllib.error.HTTPError as e:
            err = e.read().decode()
            self._json(500, {'error': f'GitHub API {e.code}: {err}'})
